### PR TITLE
fix(workspace): a template engine is not a response, a setting is not a route

### PR DESCRIPTION
**Pull request by ofri-peretz**
## Summary

Found by **rescanning the same targets after [#692](https://github.com/ofri-peretz/eslint/pull/692) shipped**, which is the point of rescanning. Two more rules deciding on a name.

### `browser-security/require-csp-headers` matched `render` on any receiver

```ts
function createRenderer(templatePath) {
  return (props, nunjucksEnv) => nunjucksEnv.render(templatePath, { params: props });
}
```

`nunjucksEnv.render(template, data)` **returns a string**. So do mustache, ejs, handlebars and ReactDOMServer. Only `res.render(view)` emits a response.

On `ministryofjustice/hmpps-arns-assessment-platform-ui` that was **31 reports** — every Nunjucks component module and every component test — in a repository that already sets a nonce-based CSP in `setUpWebSecurity.ts`. Asking a template helper to set an HTTP header is asking for something it cannot do, which is the shape of a finding nobody can action.

Gated on the receiver, and the rule now takes `skipTestFiles`: **29 of the 31 were in `*.test.ts`**, where rendering a template to assert on its markup is not a route serving a document.

**31 → 3.**

### `secure-coding/no-missing-authentication` treated `app.get('port')` as a route

```ts
const server = app.listen(app.get('port'), () => { … })
```

Express overloads the name: one argument reads a setting, two or more registers a route. A port number was being reported as an unauthenticated endpoint, as was the log line beside it.

A route always has a handler after its path, so the arity test is exact rather than a heuristic.

## Test plan

- [x] `eslint-plugin-browser-security` — 2,917 passed, coverage back to 100%.
- [x] `eslint-plugin-secure-coding` — 3,511 passed (the `redos-oracle` failures here are `recheck` missing from my local install, pre-existing and unrelated).
- [x] `rule-audit:gate --changed` — no regressions. It caught two things on the way and both were right: a baked-in word list with no option (`unconfigurable-vocabulary`) and then a schema option with no declared default.
- [x] **Both fixes verified against the file that produced them**, then the whole repository rescanned: MoJ went 105 → 75 findings across the two changes.
- [x] False-negative guards in the same diff: `res.render`, `reply.render` (Fastify) and `ctx.res.render` all still report; `app.route(path).get(handler)` still reports — its existing lock caught the first version of the arity fix, which did not exempt the chained form.

## Notes

The receiver list is an **option**, not a constant. The ratchet insisted and was right: a house convention that calls it `httpRes` should configure the list rather than lose the finding. The default lives in `defaultOptions` and the schema and nowhere else — as a module const it was an unreachable word list, and as a `??` fallback beside a named default it was an unreachable branch.

`no-missing-authentication` still reports 17 times on that repo, on the sign-in routes and health checks that must be public. That is a policy question rather than a bug, and a separate piece of work.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Reduced false positives when detecting missing CSP headers in template-engine render calls.
  * Added support for configurable response receiver names and skipped test files during CSP checks.
  * Improved authentication detection by distinguishing route registration from settings access and other non-endpoint calls.
  * Preserved detection for chained routes and supported additional router configurations.

* **Tests**
  * Expanded coverage for template engines, response objects, authentication patterns, router behavior, and edge cases.

* **Chores**
  * Prepared patch releases for both ESLint plugins.

### CODE REVIEW

File: packages/eslint-plugin-browser-security/src/rules/require-csp-headers/index.ts [hunk 1]
```diff
@@ -439,6 +505,33 @@ export const requireCspHeaders = createRule<RuleOptions, MessageIds>({
...
+      const callee = node.callee as TSESTree.MemberExpression;
+      let object: TSESTree.Node = callee.object;
+      // `this.res.render(…)` / `ctx.res.render(…)` — take the last segment.
+      if (object.type === 'MemberExpression' && !object.computed) {
+        object = object.property;
+      }
+      return (
+        object.type === 'Identifier' &&
+        responseReceivers.has(object.name.toLowerCase())
+      );
```
[the remark this answers is not in the document]
Comment (ofri-peretz) [reply packages/eslint-plugin-browser-security/src/rules/require-csp-headers/index.ts:553]: This is the same defect the PR is about, one level up — deciding on a name without trying to resolve it. Fixed.

`receiverLooksLikeResponse` now calls `resolveInitializer` first and believes the declaration over the name. Anything with a visible initialiser in the file is a renderer given a short name, because a response object is produced by a framework — it is never declared as a local alias of something else. Your case is now a valid test:

```js
const res = nunjucksEnv;
const html = res.render('index');
```

The name decides only when the identifier resolves to nothing local, and that is the case that carries the rule: `(req, res) => …` binds `res` as a **parameter**, and a parameter has no initialiser to inspect. That is why the list stays as a fallback rather than being removed — resolution answers the aliased form, and nothing answers the parameter form.

File: packages/eslint-plugin-browser-security/src/rules/require-csp-headers/a-template-engine-is-not-a-response.test.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      { code: `function handler(ctx) { ctx.res.render('index'); }`, errors: 1 },
+      // A house convention that names it something else configures the list
+      // rather than losing the finding.
+      {
+        code: `function handler(httpRes) { httpRes.render('index'); }`,
+        options: [{ responseReceivers: ['httpRes'] }],
+        errors: 1,
+      },
+    ],
+  },
```
[the remark this answers is not in the document]
Comment (ofri-peretz) [reply packages/eslint-plugin-browser-security/src/rules/require-csp-headers/a-template-engine-is-not-a-response.test.ts:84]: Correct — no case set a filename, so `skipTestFiles` was decorative in the suite. Added, with a real response object so it isolates test-file skipping from receiver filtering:

```js
{
  code: `app.get('/', (req, res) => { res.render('index'); });`,
  filename: 'component.test.ts',
}
```

That same code without the filename is an `invalid` case two lines down, so the pair pins the skip rather than the receiver.

File: packages/eslint-plugin-secure-coding/src/rules/no-missing-authentication/no-missing-authentication.test.ts [hunk 3]
```diff
@@ -620,202 +709,240 @@ describe('option: testFilePattern', () => {
...
+      // `require("express")()` - the initialiser IS a CallExpression, but its
+      // callee is another call, so no factory name can be read off it. The
+      // fallback is the binding NAME, and `gw` carries no router word: this is
+      // a documented CJS false negative on the router side, kept here only to
+      // pin the branch.
       {
-        code: 'const gateway = express(); gateway.get("/admin/secrets", (req, res) => {});',
+        code: 'const server = require("express")(); server.get("/admin/a", (req, res) => {});',
         errors: [{ messageId: 'missingAuthentication' }],
       },
```
[the remark this answers is not in the document]
Comment (ofri-peretz) [reply packages/eslint-plugin-secure-coding/src/rules/no-missing-authentication/no-missing-authentication.test.ts]: Right on both counts — `server` is in `ROUTER_NAME_WORDS`, so the name fallback succeeds and the case reports, which is exactly why it is in `invalid`. The comment was describing a different snippet.

Rewritten to say what the case actually pins: the name fallback carries the CJS factory form `require("express")()`, which the initialiser walk alone cannot, because the initialiser's callee is another call and no factory name can be read off it.
